pkg/api/v1/types.go

```diff
@@ -2030,6 +2030,9 @@ type SecurityContext struct {
 
 	// RunAsUser is the UID to run the entrypoint of the container process.
 	RunAsUser *int64 `json:"runAsUser,omitempty" description:"the user id that runs the first process in the container; see http://releases.k8s.io/HEAD/docs/design/security_context.md#security-context"`
+
+	// Run the container using the host ipc mode
+	UseHostIpc *bool `json:"useHostIpc,omitempty" description:"run the container using the host ipc mode; see http://releases.k8s.io/HEAD/docs/design/security_context.md#security-context"`
```

Comment:
+1 to moving namespace sharing to Pod level. It gets confusing once we start supporting namespace sharing among containers in a pod.
Instead of sharing namespaces with host piecemeal, can we have one knob that results in the pod sharing most of the namespaces with the host? I assume this kind of a pod is already privileged and we trust it.